pkg/auth/providers/activedirectory/activedirectory_provider.go

```diff
@@ -229,9 +229,13 @@ func (p *adProvider) getDNAndScopeFromPrincipalID(principalID string) (string, s
 	return externalID, scope, nil
 }
 
-func (p *adProvider) GetUserExtraAttributes(token *v3.Token) map[string][]string {
+func (p *adProvider) GetUserExtraAttributes(userPrincipal v3.Principal) map[string][]string {
 	extras := make(map[string][]string)
-	extras["principalid"] = []string{token.UserPrincipal.Name}
-	extras["username"] = []string{token.UserPrincipal.LoginName}
+	if userPrincipal.Name != "" {
```

Comment:
These checks don't seem to be related to the issue addressed. Is it just an extra check added to not add empty data or there is more subtle reason?
Is it pretty common to have one of or both of `userPrincipal.Name`/`userPrincipal.LoginName` empty?

Reply:
yes that data could be empty, especially for kubeconfig tokens. This is to add keys only with non-empty values so that we can detect if a fallback on reading the missing data is needed in here https://github.com/rancher/rancher/pull/33268/files#diff-ccde1c52dd5ce3e2e7e9d8c62b6a7a60eb7f80d86942e367c19586fce6b70c63R191-R196